pkg/geo/geomfn/polygon.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+// MakePolygon creates a Polygon geometry from linestring and optional inner linestrings.
+// Returns errors if geometries are not linestrings.
+func MakePolygon(outer *geo.Geometry, interior ...*geo.Geometry) (*geo.Geometry, error) {
+	outerGeomT, err := outer.AsGeomT()
+	if err != nil {
+		return nil, err
+	}
+	outerRing, ok := outerGeomT.(*geom.LineString)
+	if !ok {
+		return nil, errors.Newf("argument must be LINESTRING geometries")
+	}
+	srid := outerRing.SRID()
+	coords := make([][]geom.Coord, len(interior)+1)
+	coords[0] = outerRing.Coords()
+	for i, g := range interior {
+		interiorRingGeomT, err := g.AsGeomT()
+		if err != nil {
+			return nil, err
+		}
+		interiorRing, ok := interiorRingGeomT.(*geom.LineString)
+		if !ok {
+			return nil, errors.Newf("argument must be LINESTRING geometries")
+		}
+		if interiorRing.SRID() != srid {
+			return nil, errors.Newf("mixed SRIDs are not allowed")
+		}
+		coords[i+1] = interiorRing.Coords()
+	}
```

Comment:
one more minor detail: can you also add a test to make sure linestrings are closed with at least four points, e.g.

```
otan=# select st_makepolygon('LINESTRING(0 0, 1 0)');
ERROR:  lwpoly_from_lwlines: shell must have at least 4 points
otan=# select st_makepolygon('LINESTRING(0 0, 1 0, 1 1, 0.1 0.1)');
ERROR:  lwpoly_from_lwlines: shell must be closed
```

Reply:
Added a check for number of points and closure. Thanks! Not sure if this is only Postgis behavior or `geom` should also implement this check as well.

Reply:
Probably, but we'll make do with this for now :)

Reply:
Filed #51074.